### ISSUE TITLE
[Bug]: Error: Failed to extract n-token decipher algorithm

### ISSUE THREAD

**Issue by daviddoji** [FreeTubeApp/FreeTube#5502]
### Guidelines

- [X] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have encountered this bug in the [official downloads of FreeTube](https://github.com/FreeTubeApp/FreeTube#official-downloads).
- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the bug report I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.
- [X] This issue contains only one bug.

### Describe the bug

Trid to play any video and got the following message:
Error: Failed to extract n-token decipher algorithm

### Expected Behavior

Video reproduces

### Issue Labels

content not loading

### FreeTube Version

v0.21.2 Beta

### Operating System Version

Archlinux, up-to-date

### Installation Method

AUR (Unofficial)

### Primary API used

Local API

### Last Known Working FreeTube Version (If Any)

v0.21.2 Beta

### Additional Information

This morning, it did work :shrug: and system was not updated since.

### Nightly Build

- [ ] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by Zephiii** [comment]
Seems to have randomly started happening, like that issue from the earlier version last week.

**Comment by XenithXemaku** [comment]
Same thing here, i keep getting (in order):
`Local API Error:
Error: Failed to extract n-token decipher algorithm`
`Falling back to Invidious API`
`Invidious API Error
SyntaxError: Unexpected end of JSON input`

**Comment by darktohka** [comment]
Same error here.

**Comment by M-Hamano16** [comment]
yes same thing here

**Comment by CharaDr33murr** [comment]
Guys, just upvote the post. The issue seems to be similar to the one that happened recently - likely YouTube cracking down. Let's not spam the comments.

**Comment by darktohka** [comment]
Using the main branch of https://github.com/LuanRT/YouTube.js solves the issue on nightly.

So maybe we need to wait for a new version release of the library?

**Comment by darktohka** [comment]
To try the nightly branch of YouTube.js: edit the package.json to reflect

```
    "youtubei.js": "git+https://github.com/darktohka/YouTube.js.git"
```

There are build errors on the current main branch of YouTube.js, until the fixes are merged feel free to use my fork to test whether the new version fixes the issue.

**Comment by birb7** [comment]
> To try the nightly branch of YouTube.js: edit the package.json to reflect
> 
> ```
>     "youtubei.js": "git+https://github.com/darktohka/YouTube.js.git"
> ```
> 
> There are build errors on the current main branch of YouTube.js, until the fixes are merged feel free to use my fork to test whether the new version fixes the issue.

how do i even build freetube? theres no cmake or build instructions anywhere.

**Comment by absidue** [comment]
> Using the main branch of https://github.com/LuanRT/YouTube.js solves the issue on nightly.
> 
> 
> 
> So maybe we need to wait for a new version release of the library?

The actual fix is in an open pull request on that repo, so they definitely won't do a release until that is finished and merged.

**Comment by absidue** [comment]
> To try the nightly branch of YouTube.js: edit the package.json to reflect
> 
> 
> 
> ```
> 
>     "youtubei.js": "git+https://github.com/darktohka/YouTube.js.git"
> 
> ```
> 
> 
> 
> There are build errors on the current main branch of YouTube.js, until the fixes are merged feel free to use my fork to test whether the new version fixes the issue.

As mentioned on your YouTube.js bug reports, the build errors don't happen if you build it correctly (clone YouTube.js, build it and add the built version to FreeTube).

**Comment by birb7** [comment]
> > To try the nightly branch of YouTube.js: edit the package.json to reflect
> > ```
> > 
> >     "youtubei.js": "git+https://github.com/darktohka/YouTube.js.git"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > There are build errors on the current main branch of YouTube.js, until the fixes are merged feel free to use my fork to test whether the new version fixes the issue.
> 
> As mentioned on your YouTube.js bug reports, the build errors don't happen if you build it correctly (clone YouTube.js, build it and add the built version to FreeTube).

how would i go about doing that? do i have to add the built youtube.js, and then build freetube? or does it get put into a certain spot where freetube stores data? and where are the instructions to build freetube (if needed)?

**Comment by Robert-CMDedu** [comment]
Happening on the appimage arch

**Comment by JEsa44** [comment]
Same here. In fact, a number of issues over the past week especially.

**Comment by absidue** [comment]
We are aware of the problem that YouTube has intentionally broken stuff again, everyone is affected, so please don't comment unless you have something to add that hasn't already been said.

**Comment by absidue** [comment]
@birb7 How to build is mentioned in the docs and yes you won't find a CMake file because this is a JavaScript project.

https://docs.freetubeapp.io/development/building-from-source

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
[Upstream fix is being tested as we speak](https://github.com/LuanRT/YouTube.js/pull/713)

Temporary workaround, for the best viewing experience I recommend doing the following things:

- Set Primary API to Invidious in the General Settings
- Select an Instance that is near to you, see https://api.invidious.io/. Make sure to pick one that has a check mark in the API field. Also make sure you start typing the instance name in the Current Invidious Instance field under the General Settings with `https://` and it should list instances as suggestions.
- Disable Fallback to Non-Preferred Backend on Failure in the General Settings
- Disable Proxy Videos Through Invidious in the Player settings

> [!NOTE]
> Some Invidious instances throws the error you see in https://github.com/FreeTubeApp/FreeTube/issues/5489. You should try other Invidious instances when this occurs. It could be difficult to find a working instance because some Invidious instances are IP blocked by YouTube.

**Comment by angelfire76** [comment]
I have the same problem in almost every instance. Is there any working instance? After each update, it works properly for a week and then starts giving API errors.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
UPDATE:

Upstream fix has been merged.
[We are currently testing the fix](https://github.com/FreeTubeApp/FreeTube/pull/5507).

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
UPDATE:

Fix has successfully passed our testing. Hotfix release will be coming soon!

**Comment by izmyname** [comment]
> UPDATE:
> 
> Fix has successfully passed our testing. Hotfix release will be coming soon!

Nice! Can't wait for the appimage update.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
UPDATE: 

We are facing some technical difficulties with GitHub. We aim to resolve it ASAP so we can provide the hotfix release. Thank you all for being patient!

**Comment by ACR-Jeff** [comment]
The latest nightly has been working great for the last 12 hours for me. 
https://github.com/FreeTubeApp/FreeTube/actions/runs/10198632871

**Comment by absidue** [comment]
@konomikitten The flatpak will get updated when the release happens, there is no nightly flatpak.

**Comment by konomikitten** [comment]
> @konomikitten The flatpak will get updated when the release happens, there is no nightly flatpak.

So my understanding is if people use the flatpak version of FreeTube and YouTube breaks it we'll wait longer for it to be working again than people using the AppImage?

**Comment by absidue** [comment]
No, that is incorrect, there hasn't been a release yet, the stable/release AppImage is the same as the flatpak release.

What was linked above are the nightly/bleeding edge builds, which are created automatically from the lastest changes on the development branch. They are only meant for people that want to help test changes before they are released. They aren't intended for every day use and we recommend doing frequent backups if you do decide to use them, as the nightly builds get a lot less testing than the release builds.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
UPDATE:

The issues we were facing earlier are resolved. Release incoming!

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
UPDATE:

v0.21.3 has been released! You can download the latest version using the following methods:
- [GitHub release page](https://github.com/FreeTubeApp/FreeTube/releases/tag/v0.21.3-beta)
- [Through our website](https://freetubeapp.io/#download)

Thank you for being patient and happy watching 📺!

**Comment by exefile2** [comment]
> To try the nightly branch of YouTube.js: edit the package.json to reflect
> 
> ```
>     "youtubei.js": "git+https://github.com/darktohka/YouTube.js.git"
> ```
> 
> There are build errors on the current main branch of YouTube.js, until the fixes are merged feel free to use my fork to test whether the new version fixes the issue.

What do you mean by YouTube.js? Can anyone please explain me?
